plugins/gossip/gossip.go

```diff
@@ -60,15 +62,42 @@ func createManager(lPeer *peer.Local, t *tangle.Tangle) *gossip.Manager {
 		libp2p.NATPortMap(),
 	)
 	if err != nil {
-		Plugin.LogFatalf("Could create libp2p host: %s", err)
+		Plugin.LogFatalf("Couldn't create libp2p host: %s", err)
 	}
-
-	return gossip.NewManager(libp2pHost, lPeer, loadMessage, Plugin.Logger())
+	var opts []gossip.ManagerOption
+	if Parameters.MessagesRateLimit != (messagesLimitParameters{}) {
+		Plugin.Logger().Infof("Initializing messages rate limiter with the following parameters: %+v",
+			Parameters.MessagesRateLimit)
+		mrl, mrlErr := ratelimiter.NewPeerRateLimiter(
+			Parameters.MessagesRateLimit.Interval, Parameters.MessagesRateLimit.Limit,
+			Plugin.Logger().With("rateLimiter", "messagesRateLimiter"),
+		)
+		if mrlErr != nil {
+			Plugin.LogFatalf("Failed to initialize messages rate limiter: %+v", mrlErr)
+		}
+		opts = append(opts, gossip.WithMessagesRateLimiter(mrl))
+	}
+	mgr, err := gossip.NewManager(libp2pHost, lPeer, loadMessage, Plugin.Logger(), opts...)
+	if err != nil {
+		Plugin.LogFatalf("Couldn't create gossip manager: %+v", err)
+	}
+	return mgr
 }
 
 func start(ctx context.Context) {
 	defer Plugin.LogInfo("Stopping " + PluginName + " ... done")
-
+	if mrl := deps.GossipMgr.MessagesRateLimiter(); mrl != nil {
+		setLimitClosure := events.NewClosure(func(ev *tangle.SyncChangedEvent) {
+			if ev.Synced {
+				mrl.SetLimit(Parameters.MessagesRateLimit.Limit)
+			} else {
+				mrl.SetLimit(Parameters.MessagesRateLimit.DuringSyncLimit)
+			}
+		})
+		deps.Tangle.TimeManager.Events.SyncChanged.Attach(setLimitClosure)
+		defer deps.Tangle.TimeManager.Events.SyncChanged.Detach(setLimitClosure)
```

Comment:
If we get rid of the separate `DuringSyncLimit` this is not needed anymore. Maybe even the possibility to change the limit during runtime is not needed anymore. But since the latter it's coded, we could just leave it.